### PR TITLE
Use startsWith and endsWith when possible.

### PR DESCRIPTION
**Pull request by XhmikosR**
Makes the intention clearer.

TODO:

* [x] figure out how to handle the Vinyl test failure

### CODE REVIEW

File: src/file.js [hunk 1]
```diff
@@ -35,11 +35,15 @@ function normalizePath(str) {
 
 /**
  * Check whether a resource is external or not
- * @param {string} href Path
+ * @param {string|Vinyl} href Path
```
Comment (XhmikosR) [inline src/file.js]: Not super familiar with jsdoc so please confirm this is right

File: src/file.js [hunk 2]
```diff
@@ -35,11 +35,15 @@ function normalizePath(str) {
...
  * @returns {boolean} True if the path is remote
  */
 function isRemote(href) {
-  return /(^\/\/)|(:\/\/)/.test(href) && !href.startsWith('file:');
+  if (isVinyl(href)) {
+    return false;
```
Comment (XhmikosR) [inline src/file.js]: I ended up adding a separate check for `Vinyl`. I'm not sure if the test is right; if it is then this check makes sense. If not, we should change the test @bezoerb

Comment (bezoerb) [reply src/file.js]: The `isRemote` function shouldn't intentionally receive a vinyl object. Until we find out where the vinyl is coming from the test should be something like this: 
```js
if (isVinyl(href)) {
   // Check custom attribute set in the vinylize function
  return Boolean(href.remote)
}
```

File: src/file.js (hunk 1 above)
Comment (bezoerb) [reply src/file.js]: Shouldn't be ;)
Do you have a hint to start debugging where the Vinyl object is coming from?

Comment (XhmikosR) [reply src/file.js]: It happens in one of the vinyl tests. I'll revert this part so that we get more info.

**Comment by XhmikosR** [comment]
@bezoerb please check now, you should see the failure

**Pull request closed without merging**